test/unit/modules/delegates.js

```diff
@@ -489,6 +490,172 @@ describe('delegates', () => {
 				});
 			});
 		});
+
+		describe('getDelegateKeypairForCurrentSlot', () => {
+			let delegates;
+			let __private;
+
+			before(done => {
+				delegates = library.rewiredModules.delegates.__get__('self');
+				__private = library.rewiredModules.delegates.__get__('__private');
+
+				__private.keypairs = {
+					// genesis_1 delegate keypairs
+					'9d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f': {
+						publicKey: new Buffer.from(
+							'9d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f',
+							'hex'
+						),
+						privateKey: new Buffer.from(
+							'b092a6664e9eed658ff50fe796ee695b9fe5617e311e9e8a34eb340eb5b831549d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f',
+							'hex'
+						),
+					},
+					// genesis_2 delegate keypairs
+					'141b16ac8d5bd150f16b1caa08f689057ca4c4434445e56661831f4e671b7c0a': {
+						publicKey: new Buffer.from(
+							'141b16ac8d5bd150f16b1caa08f689057ca4c4434445e56661831f4e671b7c0a',
+							'hex'
+						),
+						privateKey: new Buffer.from(
+							'e896941f49df461e34d87cbe66c9cf35abcc58a25e4494c97a3f781b9e77e83c141b16ac8d5bd150f16b1caa08f689057ca4c4434445e56661831f4e671b7c0a',
+							'hex'
+						),
+					},
+					// genesis_3 delegate keypairs
+					'3ff32442bb6da7d60c1b7752b24e6467813c9b698e0f278d48c43580da972135': {
+						publicKey: new Buffer.from(
+							'3ff32442bb6da7d60c1b7752b24e6467813c9b698e0f278d48c43580da972135',
+							'hex'
+						),
+						privateKey: new Buffer.from(
+							'd4b9f54978f789c1a291f948d32f19039d54962b5b4c14d314c1521f2975b9be3ff32442bb6da7d60c1b7752b24e6467813c9b698e0f278d48c43580da972135',
+							'hex'
+						),
+					},
+				};
+
+				done();
+			});
+
+			it('should return genesis_1 keypair for slot N where (N % 101 === 35) in the first round', done => {
+				// For round 1, delegates genesis_1, genesis_2 and genesis_3 should forge for slots 35, 53 and 16 respectively.
+				const currentSlot = 35;
+				const round = 1;
+
+				delegates.generateDelegateList = (round, source, cb) => {
+					cb(null, delegatesRoundsList[round]);
+				};
+
+				__private.getDelegateKeypairForCurrentSlot(
+					currentSlot,
+					round,
+					(err, keyPair) => {
+						expect(err).to.be.null;
+						expect(keyPair).to.have.keys('publicKey', 'privateKey');
+						expect(keyPair.publicKey.toString('hex')).to.equal(
+							'9d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f'
```

Comment:
Better would be store those keys explicitly to variable e.g. `genesis_1_publicKey` and use those in expectations. This will make expectations more readable.